spec/lib/smart_name_spec.rb

```diff
@@ -238,4 +246,22 @@
     end
   end
 
+  describe "namespaced namse" do
+    let(:root_part) { 'RootNamespace' }
+    let(:plus_a) { root_part + '+A' }
+    let(:subspace) { 'PathElement+SubNamespace' }
+    let(:plus_sub) { "#{root_part}+#{subspace}" }
+    let(:plus_bc)  { plus_sub + '+B+C' }
+
+    it 'works in namespaces' do
+      root_part.to_name.rootspace.should be_true
+      plus_a.to_name.rootspace.should be_false
+      plus_sub.to_name.rootspace.should be_false
+      root_part.to_name.namespaced.should == [root_part]
+      plus_a.to_name.namespaced.should == [root_part, 'A']
+      plus_sub.to_name.namespaced.should == [root_part, subspace]
+      plus_bc.to_name.namespaced.should == [root_part, subspace, 'B+C']
```

Comment:
Each part returned will be relative to it's namespace, and all but the last must name another namespace, so if we changed the semantics the the / // /// stuff, these would all have a single / on the front.